command.ts

```diff
@@ -657,41 +689,45 @@ XRANGE somestream - +
    * @param maxlen 
    */
   xtrim(key: string, maxlen: XMaxlen): Promise<Integer>;
+
   // SortedSet
-  bzpopmin(key: string | string[], timeout: number): Promise<
-    [BulkString, BulkString, BulkString] | []
-  >;
-  bzpopmax(key: string | string[], timeout: number): Promise<
-    [BulkString, BulkString, BulkString] | []
-  >;
+  bzpopmin(
+    timeout: number,
+    ...keys: string[]
+  ): Promise<[BulkString, BulkString, BulkString] | []>;
+  bzpopmax(
+    timeout: number,
+    ...keys: string[]
+  ): Promise<[BulkString, BulkString, BulkString] | []>;
   zadd(
     key: string,
     score: number,
     member: string,
-    opts?: {
-      nxx?: "NX" | "XX";
-      ch?: boolean;
-      incr?: boolean;
-    },
+    opts?: { mode?: "NX" | "XX"; ch?: boolean },
   ): Promise<Integer>;
   zadd(
     key: string,
-    score_members: [number, string][],
```

Comment:
Ditto

Reply:
Do you think `[number, string][]` or `(number|string)[]` is better here? The first one syntax feels awkward and I think most users would rather write object form. The second feels more Redis'y, just list everything out.

Reply:
Personally, I'd like to support both forms (array and object):

- In JS, it is easy to convert an array to an object and vice versa.
- Since many libraries only support the array form, I think users might be more familiar with it.
  - `ioredis` only supports the object form.
  - `node-redis` only supports the object form.
  - `go-redis` only supports the array form.
  - `redis-py` **only supports the object form**.
  - `redis-rb` only supports the array form.

Reply:
Yes I'm not against it. I was asking if you think `[[1, "one"], [2, "two"]]` or `[1, "one", 2, "two"]` is better. I think the second one is easier to write 😃

Reply:
Oh, I think `[[1, "one"], [2, "two"]]` is better in terms of type safety :smile:

Reply:
Also in `hset` we have `...field_values: string[]`. Should it be `[string, string][]` to be consistent with the above (explicitly requires a pair of values)?

Reply:
How about `...field_values: [string, string][]`?